### ISSUE TITLE
gw_iface sometimes returns (null)

### ISSUE THREAD

**Issue by Ambrevar** [brndnmtthws/conky#461]
The following prints "wlp3s0", my wifi interface:

```
$gw_iface
```

But this always prints "0.0":

```
${downspeedf $gw_iface}
```

The following trick does it:

```
${eval $${downspeedf $gw_iface}}
```

Now the interesting this is that the `eval` trick seems to "kill" the gw_iface variable since afterwards it always returns `(null)`.
The following returns "123.4 (null)".

```
${eval $${upspeedf $gw_iface}} \
$gw_iface
```

Bug?  Or am I missing something?

**Comment by lasers** [comment]
`gw_iface`, `eval`, `downspeedf eno1`
```
eno1 // 0.0 // 0.1
eno1 // 0.0 // 0.2
eno1 // 0.0 // 0.3
eno1 // 0.0 // 0.2
eno1 // 0.0 // 0.1
eno1 // 0.0 // 0.2
eno1 // 0.0 // 0.2
eno1 // 0.0 // 0.1
eno1 // 0.0 // 0.9
eno1 // 0.0 // 309.1
eno1 // 0.0 // 731.8
eno1 // 0.0 // 902.8
eno1 // 0.0 // 2823.6
eno1 // 0.0 // 3104.6
eno1 // 0.0 // 2933.5
eno1 // 0.0 // 3930.7
eno1 // 0.0 // 3671.4
eno1 // 0.0 // 1914.0
```
>@Ambrevar The following trick does it:
>```
>${eval $${downspeedf $gw_iface}}
>```

That following trick looks like a placebo as it gave me this.

`gw_iface`, `eval`, `downspeedf eno1`
```
eno1 // 0.01 // 0.1                                                                           
eno1 // 0.01 // 0.2                                                                           
eno1 // 0.01 // 0.2                                                                           
eno1 // 0.01 // 0.2                                                                           
eno1 // 0.01 // 184.2                                                                         
eno1 // 0.01 // 335.7                                                                         
eno1 // 0.01 // 223.4                                                                         
eno1 // 0.01 // 138.6                                                                         
eno1 // 0.01 // 142.6                                                                         
eno1 // 0.01 // 140.9                                                                         
eno1 // 0.01 // 197.3                                                                         
eno1 // 0.01 // 138.8  
```
I think it's a bug too.

**Comment by su8** [comment]
With the current code, once the variables are parsed they are cleared out:

https://github.com/brndnmtthws/conky/blob/c771698d5a171cb15ef44d042f42f9c866537934/src/conky.cc#L904

**Comment by lasers** [comment]
@su8 Is that for `${eval xxx}`? `${downspeedf $gw_iface}` is the problem here as it should behave similar to the `${pid_nice ${top pid 1}}`

**Comment by su8** [comment]
@lasers with the current code

https://github.com/brndnmtthws/conky/blob/7bca0ca8819f03883affa5fba9e8506d8c2c5f29/src/net_stat.cc#L110

parses the argument which must not be a variable

**Comment by lasers** [comment]
Closed by #571 :ok_hand:
